core/src/test/scala/unit/kafka/server/ReplicaAlterLogDirsThreadTest.scala

```diff
@@ -147,7 +167,7 @@ class ReplicaAlterLogDirsThreadTest {
     mockFetchFromCurrentLog(tid1p0, fencedRequestData, config, replicaManager, fencedResponseData)
 
     val endPoint = new BrokerEndPoint(0, "localhost", 1000)
-    val thread = new ReplicaAlterLogDirsThread(
+    val thread = createReplicaAlterLogDirsThread(
```

Comment:
Why not just include the new `LeaderEndpoint` parameter directly here, instead of introducing a helper?
i.e.

```
val thread = new ReplicaAlterLogDirsThread(
      "alter-logs-dirs-thread",
      new LocalLeaderEndPoint(replicaManager),
      sourceBroker = endPoint,
      brokerConfig = config,
      failedPartitions = failedPartitions,
      replicaMgr = replicaManager,
      quota = quotaManager,
      brokerTopicStats = new BrokerTopicStats)
```